storage-transcribe-audio/functions/src/transcribe-audio.ts

```diff
@@ -0,0 +1,138 @@
+import { SpeechClient } from "@google-cloud/speech";
+import { google } from "@google-cloud/speech/build/protos/protos";
+import { ObjectMetadata } from "firebase-functions/v1/storage";
+import * as ffmpeg from "fluent-ffmpeg";
+import * as util from "util";
+
+import * as logs from "./logs";
+import config from "./config";
+
+export async function transcribeAudio(
+  client: SpeechClient,
+  object: ObjectMetadata
+): Promise<void> {
+  const request = {
+    config: {
+      encoding: "LINEAR16" as const,
+      sampleRateHertz: 48000,
+      languageCode: config.languageCode,
+      model: config.model,
+    },
+    audio: {
+      uri: `gs://${object.bucket}/${object.name}`,
+    },
+  };
+
+  const response = await transcribeLongOrUnknownLengthAudio(client, request);
+  logs.receivedLongRunningRecognizeResponse(response);
+  const transcript = response.results?.map(
+    (result) => result?.alternatives?.[0].transcript
+  );
+
+  if (!isStringArray(transcript)) {
+    logs.undefinedResultsTranscript(transcript);
+  } else {
+    logs.logResponseTranscript(transcript);
+  }
+
+  // TODO(reao): Handle user configs
+  // TODO(reao): Add tests
+  // TODO(reao): Handle files of known short length
+  // TODO(reao): Transcode files into lossless format (without upsampling)
+  // TODO(reao): Handle video files
+}
+
+type TranscodeAudioResult = TranscodeAudioSuccess | TranscodeAudioFailure;
```

Comment:
Let's maybe put all of these types and interfaces near the top.

Reply:
SGTM; I'll let you know when I've done that

Reply:
Moved the interfaces to their own file.